openassessment/data.py

```diff
@@ -498,6 +498,56 @@ def collect_ora2_data(cls, course_id):
         ]
         return header, rows
 
+    @classmethod
+    def collect_ora2_summary(cls, course_id):
+        """
+        Query database for aggregated ora2 summary data.
+
+        Args:
+            course_id (string) - the course id of the course whose data we would like to return
+
+        Returns:
+            A tuple containing two lists: headers and data.
+
+            headers is a list containing strings corresponding to the column headers of the data.
+            data is a list of lists, where each sub-list corresponds to a row in the table of all the data
+                for this course.
+
+        """
+
+        items = AssessmentWorkflow.objects.filter(course_id=course_id)
+
+        rows = []
+        for aw in items:
+            statuses = aw.status_details()
+            submission_dict = sub_api.get_submission_and_student(aw.submission_uuid)
+
+            row = [
```

Comment:
There's some additional columns requested by the client, which should be added to the `AssessmentWorkflow.status_details()` output so it can be reported here:

* `unit_name`: the Open Responses tab can do this, so I believe that means we can too.  cf [`ora_blocks_listing_view()`](https://github.com/edx/edx-ora2/blob/028bd31f054533223e8a9df3bccd5c036e910d61/openassessment/xblock/openassessmentblock.py#L425), should be moved into an API.
* `assessment_name`: ditto ^
* `num_peers_assessed`: make [this logic from the `api.peer.has_finished_required_evaluating()`](https://github.com/edx/edx-ora2/blob/76e5a3c614a03af845229dda583380ac06636907/openassessment/assessment/api/peer.py#L478-L479) accessible directly from the `peer.api`, and include this data in the `AssessmentWorkflow.status_details()` output for the `peer` step.
* `num_assessed_by_peers`: make [this logic from the `api.peer.assessment_is_finished()`](https://github.com/edx/edx-ora2/blob/76e5a3c614a03af845229dda583380ac06636907/openassessment/assessment/api/peer.py#L86-L90) accessible directly from the `peer.api`, and include this in the  for the `peer` step.
* `peers_assessed_done`: this wasn't requested in the original report, but since we already have an API call for [has_finished_required_evaluating()](https://github.com/edx/edx-ora2/blob/76e5a3c614a03af845229dda583380ac06636907/openassessment/assessment/api/peer.py#L443), it's easy to add a boolean column for this.

Reply:
> unit_name: the Open Responses tab can do this, so I believe that means we can too. cf ora_blocks_listing_view(), should be moved into an API.

Currently the display names shown in the Open Responses tab are given to the function via the context dict from edx-platform, so that part isn't as easy unfortunately. 

> peers_assessed_done: this wasn't requested in the original report, but since we already have an API call for has_finished_required_evaluating(), it's easy to add a boolean column for this.

This should be already implemented as part of the steps complete columns.

Reply:
>> `unit_name`
> Currently the display names shown in the Open Responses tab are given to the function via the context dict from edx-platform, so that part isn't as easy unfortunately.

Ok, then we just need to make sure the client knows this limitation.

>> `peers_assessed_done`
> This should be already implemented as part of the steps complete columns? or does the complete status for the 'peer' step mean that the student has both received and sent enough?

Yep, the latter..  for an ORA problem to be fully "complete", a student must complete and receive a certain number of reviews.  So it would be good to add this metric to this report.

Reply:
The ["arbitrary and ignored" number you mentioned here](https://github.com/edx/edx-ora2/pull/1250/files#diff-9da7828718b3af144f7a2603e217d725R233) is the key -- if you can pass in the actual `required_assessments` number when you call `has_finished_required_evaluating`, then that `done` number should be the `peers_assessed_done` requested here.

Reply:
I can't work out how to get the `required_assessments` variable - that seems to be injected into the xblock context at runtime. However, I discovered I can use the two keys in the status dictionary - 'complete' corresponds to whether the student has assessed enough peers and 'graded' corresponds to whether the student has been assessed by enough peers.

Reply:
👍

Reply:
I added all the 'complete' and 'graded' columns in the report for completeness. This is the most generic. The only issue may be that the names aren't exactly clear and will need documenting (as in comment above).